include/ck/tensor_operation/gpu/block/reduction_functions_blockwise.hpp

```diff
@@ -80,21 +91,28 @@ struct PartitionedBlockwiseReduction
                 index_t offset2 = block_buf_desc_m_k.CalculateOffset(thread_cluster_idx +
                                                                      make_tuple(0, indOffset));
 
-                AccDataType opData1 = type_convert<AccDataType>(block_buffer[offset1]);
-                AccDataType opData2 = type_convert<AccDataType>(block_buffer[offset2]);
+                AccDataType opData1 = work_buffer[offset1];
```

Comment:
Why not using type_convert?

Reply:
So far, it is assumed the work_buffer should have the same data type as AccDataType

Reply:
Please add static_assert in this PR

Reply:
This is done in [commit 8648](https://github.com/ROCmSoftwarePlatform/composable_kernel/pull/152/commits/ded71f1f4c1135ffb04923eda103a0adfb3c8648)